python-sdk/src/astro/databases/base.py

```diff
@@ -744,11 +745,14 @@ def check_schema_autodetection_is_supported(self, source_file: File) -> bool:
             source_file.location.location_type
         )
 
-        source_filetype = create_file_type(
-            path=source_file.path, filetype=source_file.type.name
+        source_filetype = (
+            source_file
+            if isinstance(source_file.type, FileTypeConstants)
+            else create_file_type(path=source_file.path, filetype=source_file.type)  # type: ignore
         )
+
         is_source_filetype_supported = (
-            (source_filetype in filetype_supported.get("filetype"))  # type: ignore
+            (source_filetype.type.name in filetype_supported.get("filetype"))  # type: ignore
```

Comment:
Can you add a test to avoid regression please?

Reply:
Added the test https://github.com/astronomer/astro-sdk/pull/872/commits/d572c8c9b5c1749b6bfe96e3bc61bb62f3060cb6